src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -308,27 +308,42 @@ export class RunParametersAction extends TooltipFromLabelAction {
 		const editor = this._notebookService.findNotebookEditor(context);
 		// Set defaultParameters to the parameter values in parameter cell
 		let defaultParameters = new Map<string, string>();
+		let emptyParameterCell: boolean;
 		editor.cells.forEach(cell => {
 			if (cell.isParameter) {
+				// Check if cell is empty or contains just whitespace
+				const cellSource = typeof cell.source === 'string' ? [cell.source] : cell.source;
```

Comment:
We have similar logic in too many places. Is there anywhere where we're accessing the cell model from outside of that class where we want source as a string?

Not for this PR; but can you create a cleanup item? We probably have this logic in 10 different places. The cell model should either always return source as an array, or (if we need it), provide a way to return source as an array.

Reply:
As far as I know, I don't see references for wanting the source as a string. I can be sure to add this as a cleanup item / refactor.